src/pages/workspace/distanceRates/UnitSelector/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {useState} from 'react';
+import type {StyleProp, ViewStyle} from 'react-native';
+import {View} from 'react-native';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import type {UnitItemType, UnitType} from './types';
+import UnitSelectorModal from './UnitSelectorModal';
+
+type UnitSelectorProps = {
+    /** Function to call when the user selects a unit */
+    setNewUnit: (value: UnitItemType) => void;
+
+    /** Currently selected unit */
+    defaultValue: string;
+
+    /** Label to display on field */
+    label: string;
+
+    /** Any additional styles to apply */
+    wrapperStyle: StyleProp<ViewStyle>;
+};
+
+function UnitSelector({defaultValue = '', wrapperStyle, label, setNewUnit}: UnitSelectorProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+
+    const showPickerModal = () => {
+        setIsPickerVisible(true);
+    };
+
+    const hidePickerModal = () => {
+        setIsPickerVisible(false);
+    };
+
+    const updateUnitInput = (UnitItem: UnitItemType) => {
+        setNewUnit(UnitItem);
+        hidePickerModal();
+    };
+
+    const title = defaultValue ? translate(`workspace.distanceRates.units.${defaultValue as UnitType}`) : '';
+    const descStyle = title.length === 0 ? styles.textNormal : null;
+
+    return (
+        <View>
+            <MenuItemWithTopDescription
+                shouldShowRightIcon
+                title={title}
+                description={label}
+                descriptionTextStyle={descStyle}
+                onPress={showPickerModal}
+                wrapperStyle={wrapperStyle}
+            />
+            <UnitSelectorModal
+                isVisible={isPickerVisible}
+                currentUnit={defaultValue}
+                onClose={hidePickerModal}
+                onUnitSelected={updateUnitInput}
+                label={label}
+            />
```

Comment:
Oh I knew we had something like this. I looked for them but my search didn't return any results and I was kinda confused. Yes, we should just reuse them.

Reply:
I'm not really sure if its worth it. They might look the same but the common part is very small - it's just usage of `SelectionList`. On top of that `UnitPage` component has some logic specific to it's screen inside so it would have to be refactored first in order to reuse that in both places. 

What could possibly be done is extracting that SelectionList usage into separate component and call it `UnitSelector`. However I'm not fully convinced that this is worth either. We would create another component just to avoid duplication of ~10 lines (data object passed to SelectionList). Wdyt?

Reply:
I've created a new `UnitPicker` component that is used in both of these places